designs/2019-config-simplification/README.md

```diff
@@ -0,0 +1,627 @@
+* `ruledefs` - Contains definitions for rules grouped by a specific name. This replaces the `plugins` key in `.eslintrc` files and the `--rulesdir` option.
+
+The following keys are specified the same as in `.eslintrc` files:
+
+* `globals`
+* `settings`
+* `rules`
+* `parserOptions`
+
+The following keys are specified differently than in `.eslintrc` files:
+
+* `extends` - an object or array in `eslint.config.js` files (a string or string array in `.eslintrc`)
+* `parser` - an object in `eslint.config.js` files (a string in `.eslintrc`)
+* `processor` - an object in `eslint.config.js` files (a string in `.eslintrc`)
+
+Each of these keys used to require one or more strings specifying module(s) to load in `.eslintrc`. In `eslint.config.js`, these are all objects, requiring users to manually specify the objects to use.
+
+The following keys are invalid in `eslint.config.js`:
+
+* `env` - responsibility of the user
+* `overrides` - responsibility of the user
+* `plugins` - replaced by `ruledefs`
+* `root` - always considered `true`
+
+Each of these keys represent different ways of augmenting how configuration is calculated and all of that responsibility now falls on the user.
+
+#### Referencing Plugin Rules
+
+The `plugins` key in `.eslintrc` was an array of strings indicating the plugins to load, allowing you to specify processors, rules, etc., by referencing the name of the plugin. It's no longer necessary to indicate the plugins to load because that is done directly in the `eslint.config.js` file. For example, consider this `.eslintrc` file:
+
+```yaml
+plugins:
+  - react
+
+rules:
+  react/jsx-uses-react: error
+```
+
+This file tells ESLint to load `eslint-plugin-react` and then configure a rule from that plugin. The `react/` is automatically preprended to the rule by ESLint for easy reference.
+
+In `eslint.config.js`, the same configuration is achieved using a `ruledefs` key:
+
+```js
+const reactPlugin = require("eslint-plugin-react");
+
+exports.config = {
+    files: ["*.js"],
+    ruledefs: {
+        react: reactPlugin.rules
+    },
+    rules: {
+        "react/jsx-uses-react": "error"
+    }
+};
+```
+
+Here, it is the `ruledefs` that assigns the name `react` to the rules from `eslint-plugin-react`. The reference to `react/` in a rule will always look up that value in the `ruledefs` key.
+
+#### Referencing Parsers and Processors
+
+In `.eslintrc`, the `parser` and `processor` keys required strings to be specified, such as:
+
+```yaml
+plugins: ["markdown"]
+parser: "babel-eslint"
+processor: "markdown/markdown"
+```
+
+In `eslint.config.js`, you would need to pass the references directly, such as:
+
+```js
+exports.config = {
+    files: ["*.js"],
+    parser: require("babel-eslint"),
+    processor: require("eslint-plugin-markdown").processors.markdown
```

Comment:
No. The final design for the `processor` key ended up being a single entry because after processing the result is matched against all possible `files` again. See https://github.com/eslint/rfcs/tree/master/designs/2018-processors-improvements#apply-processors-using-overrides